### PR TITLE
Updating pagination to be ready for 2.1

### PR DESCRIPTION
**Pull request by zrush-mitre**

### CODE REVIEW

File: medallion/views/objects.py [hunk 1]
```diff
@@ -50,6 +49,106 @@ def get_custom_headers(api_root, id_):
...
+            new = []
+            # this may be too inefficient (i.e. O(n^2))
+            for man in manifest['objects']:
+                # versions should probably have its own seperate function
+                for check in objects['objects']:
+                    check_time = convert_to_stix_datetime(check[find_att(check)])
+                    man_time = convert_to_stix_datetime(man[find_att(man)])
+                    if check['id'] == man['id'] and check_time == man_time:
+                        new.append(check)
+                if len(new) == limit:
```
[the remark this answers is not in the document]
Comment (emmanvg) [reply medallion/views/objects.py]: (this was supposed to be from last week, I forgot to submit the review)

File: medallion/views/manifest.py [hunk 2]
```diff
@@ -28,7 +29,8 @@ def get_object_manifest(api_root, collection_id):
             api_root, collection_id, request.args, ("id", "type", "version", "spec_version"),
         )
         if manifests:
-            headers = get_custom_headers(api_root, collection_id)
+            headers = get_and_enforce_limit(api_root, collection_id, manifests)
+            # headers = get_custom_headers(api_root, collection_id)
```
Comment (clenk) [inline medallion/views/manifest.py]: This line can be removed.

File: medallion/views/objects.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zrush-mitre) [reply medallion/views/objects.py]: Yes I found that while working today. I am going to change the value of limit when no parameter is provided to be the server "max_page_size", and add a condition to the if statement to ensure that limit != len(objects["objects"])

File: medallion/views/objects.py [hunk 3]
```diff
@@ -50,6 +49,102 @@ def get_custom_headers(api_root, id_):
...
+            header values matching objects being sent in server response
+        objects:
+            Values matching server request, sorted by ascending date_added
+
+    """
+    headers = {}
+    if request.args.get('limit'):
+        limit = int(request.args['limit'])
+    else:
+        limit = current_app.taxii_config["max_page_size"]
```
Comment (emmanvg) [inline medallion/views/objects.py]: The logic here is still off. This section of the code is saying that when we don't use the `limit` parameter we should use the server configured maximum. This is not the case, if limit is not provided in a request we always return the whole result.

The max_page_size is only meant to override the provided limit if its bigger than the server maximum.

File: medallion/views/objects.py [hunk 4]
```diff
@@ -50,6 +49,102 @@ def get_custom_headers(api_root, id_):
...
+    try:
+        manifest = current_app.medallion_backend.get_object_manifest(
+            api_root, id_, None, ("id",),
+        )
+        if manifest:
```
Comment (emmanvg) [inline medallion/views/objects.py]: This block is not meant to be executed if `limit` is not provided in the request and instead can use the short version present in get_custom_headers()

Comment (zrush-mitre) [reply medallion/views/objects.py]: Are you talking about the whole function? The values still need to get sorted by date_added, which happens here.

Comment (emmanvg) [reply medallion/views/objects.py]: You could modify this function, but currently that means that objects["more"] = True would be a false statement if I made a regular request (without `limit`)

Comment (zrush-mitre) [reply medallion/views/objects.py]: After the most recent push, objects["more"] seems to work as intended, only being set as true if not all of the objects were returned. objects["more"] is returned as false when there is no limit parameter or when the limit value does not affect the total number of objects returned.

Comment (emmanvg) [reply medallion/views/objects.py]: Yep, confirmed its working for non-paginated requests and when limit == the result set.

**Pull request merged**